crates/router/src/core/payments.rs

```diff
@@ -621,10 +625,23 @@ pub async fn list_payments(
                 )
             })?;
 
-    let data: Vec<api::PaymentsResponse> = payment_intent
-        .into_iter()
-        .map(types::transformers::ForeignInto::foreign_into)
-        .collect();
+    let pi = futures::stream::iter(payment_intents)
+        .filter_map(|pi| async {
+            let pa = db
+                .find_payment_attempt_by_payment_id_merchant_id(
+                    &pi.payment_id,
+                    merchant_id,
+                    merchant.storage_scheme,
```

Comment:
Should we add a comment and say this is `olap` function call and the storage.schema is always `psql` only?